indexer/src/lib.rs

```diff
@@ -63,23 +63,55 @@ impl<'a, INDEX: Index> Indexer<'a, INDEX> {
                         match self.storage.list_all_objects().await {
                             Ok(objects) => {
                                 pin_mut!(objects);
-                                while let Some(obj) = objects.next().await {
-                                    match obj {
-                                        Ok((key, obj)) => {
-                                            log::info!("Reindexing {}", key);
-                                            if let Err(e) = self.index_doc(self.index.index(), writer.as_mut().unwrap(), &key, &obj).await {
-                                                log::warn!("(Ignored) Internal error when indexing {}: {:?}", key, e);
+
+                                let mut interval = tokio::time::interval(self.sync_interval);
+                                loop {
+                                    let tick = interval.tick();
+                                    pin_mut!(tick);
+                                    select! {
+                                        next = objects.next() => {
+                                            if let Some(obj) = next {
+                                                    match obj {
+                                                        Ok((key, obj)) => {
```

Comment:
Is this more clear? Less nesting anyway.

```suggestion
                                            match next {
                                                Some(Ok((key, obj))) => { ... },
                                                _ => break;
```

Reply:
Thanks. For some reason rustfmt doesn't touch this.

Reply:
Yeah, I thought it was just some diff formatting and got fooled 🙈